strategies/failed/src/main/java/org/arquillian/smart/testing/strategies/failed/InProjectTestReporterLoader.java

```diff
@@ -38,11 +47,12 @@
                         }
                     })
                     .map(is -> {
-                        try {
-                            return SurefireReader.loadTestResults(is);
-                        } catch (XMLStreamException e) {
-                            return null;
+                        final Optional<TestResultParser> testResultParser = javaSPILoader.onlyOne(TestResultParser.class, trp -> "junit".equals(trp.type()));
```

Comment:
Well searching for type is one, also for example searching for instanceof might be a use case, for example since services might have an empty constructor, you might need to inject some values with setter so you can do `.onlyOne(tpr -> tpr.getClass().instanceOf MyNewTestParser.class)` and then cast to desired object.